ledger/participant-integration-api/src/main/scala/platform/store/cache/StateCache.scala

```diff
@@ -73,23 +96,32 @@ private[platform] case class StateCache[K, V](cache: Cache[K, V], registerUpdate
     * (i.e. it has `validAt` highest amongst the competing updates).
     *
     * @param key the key at which to update the cache
-    * @param validAt ordering discriminator for pending updates for the same key
-    * @param eventualValue the eventual result signaling successful enqueuing of the cache async update
+    * @param fetchAsync fetches asynchronously the value for key `key` at the current cache index
     */
-  final def putAsync(key: K, validAt: Long, eventualValue: Future[V])(implicit
+  final def putAsync(key: K, fetchAsync: Long => Future[V])(implicit
```

Comment:
One the face of it if two fetches are running in parallel while the externally keyed item is being updated then either fetch could return the latest.  I assume that passing the index in some way prevents this from being a problem.

It could be worth adding a comment that explicitly defines the semantics of `fetchAsync`

Reply:
I have tried to in the description of the argument in this method's ScalaDoc. Is that clear enough or you would rather see it expanded?